### ISSUE TITLE
Moving to Azure-Samples org

### ISSUE THREAD

**Issue by bradygaster** [joergjo/contosoads-containerapp#1]
The repo looks fantastic. Great contribution, and appreciate you using one of the Microsoft company names so it'll make it easier to bring into the fold, so to speak. Adding @paulyuk from the Dapr team, as I know he'll be excited to see this content. Also adding @craigshoemaker, my partner from the docs team. 

Here's a checklist of things I'd presume we need to do to pull this in:

- [x] Decide on a repo name so we can put it at Azure-Samples/{slug-here} - @joergjo, i'd be open to your ideas on this, or we could just keep the `ContosoAds-on-AzureContainerApps` name you've got here and go with it?
- [x] @joergjo provides a brief summary for the samples page. 
- [ ] @bradygaster will facilitate the pull request to the [ACA samples page](https://docs.microsoft.com/en-us/azure/container-apps/samples)
- [ ] Get sign-off from @paulyuk and @craigshoemaker the content is ready to go. 
- [ ] We all review the pull request.
- [ ] New sample goes live. 

All good with this?

**Comment by joergjo** [comment]
LGTM. 

@bradygaster: Regarding the name: `ContosoAds-on-AzureContainerApps` is fine for me, but I'm open to other suggestions as well.

**Comment by craigshoemaker** [comment]
LGTM

@bradygaster: Are these samples being registered with the the [Azure Samples browser](https://docs.microsoft.com/samples/browse/) as well?

**Comment by bradygaster** [comment]
@craigshoemaker i don't believe so - i'd love to be educated on how to do that. Want to do one of the existing repos and send me a PR, so i can learn from what you do and tweak the others?

**Comment by joergjo** [comment]
@bradygaster: "joergjo provides a brief summary for the samples page." - where do I need to provide the summary? Or ist that by E-mail?

**Comment by bradygaster** [comment]
Sorry about the delay - was on vacation. Take a look here: https://docs.microsoft.com/en-us/azure/container-apps/samples

Essentially something like one of these paragraphs introducing the sample.

**Comment by joergjo** [comment]
Yes, I'll post something here by tomorrow.

**Comment by joergjo** [comment]
@bradygaster Does that look OK?

**Comment by bradygaster** [comment]
Yep! That looks pretty good! I will get on this tomorrow and get it pulled in. Thanks SO much! cc @craigshoemaker FYI - let's also sync up on getting these into the samples browser.

**Comment by bradygaster** [comment]
@joergjo I've got the new azure-samples repo created, and I was in the process of moving stuff over but then I realized you don't have a github actions workflow file in here. Would you prefer I set that up for you or were you planning on adding it?

**Comment by bradygaster** [comment]
I will try to get to that and get it in, and update the readme as such, and will keep everyone up-to-date here.

**Comment by joergjo** [comment]
@bradygaster I'm happy to jump back in and tackle the missing GH action workflow. I've already added some updates to bring the sample up-to-date.

**Comment by bradygaster** [comment]
Appreciate that! I have an internal email about the repo I'm trying to figure out what to do about. Something triggered a review of it - nothing on your side, just a delay is all. I'll poke the issue to see if there's any way we can move forward. Also cc @paulyuk as he'll be excited about your use of Dapr in the sample.

**Comment by joergjo** [comment]
OK, should be done by early next week.

**Comment by bradygaster** [comment]
I'll also try to push through this odd internal axle I seem to have gotten wrapped around with getting this pushed out.

**Comment by joergjo** [comment]
@bradygaster 

*should be done by early next week* said the guy who exhausted his Azure credits tow days later...

The workflow has been added. It follows the same approach as the other Azure Container Apps samples (e.g., https://github.com/Azure-Samples/dotNET-FrontEnd-to-BackEnd-on-Azure-Container-Apps), but simply deploys the Bicep files and uses my pre-built images available on Docker Hub.

**Comment by bradygaster** [comment]
@joergjo excellent thank you so much. Curious - did you sign the CLA? cc @craigshoemaker just to make sure you've signed the CLA and we're good to go there.

**Comment by joergjo** [comment]
I don't think I have (is it required for FTEs?). But how do I do that? AFAIK signing the CLA is triggered when submitting a PR, which in this case doesn't seem to apply.

**Comment by bradygaster** [comment]
OH never mind then!

**Comment by bradygaster** [comment]
Running the build now. Note - I had to make one change to the workflow file to get it to launch:

```

name: Deploy ContosoAds to Azure Container Apps

on:    
  push:
    branches:
      - deploy
    paths:
      - 'components/**'
      - 'deploy/**'
      - 'src/**'
      - '.github/workflows/**'
 ```

**Comment by joergjo** [comment]
See step 5 of the README - you have to create a secret called `DB_PWD` for PostgreSQL.

**Comment by joergjo** [comment]
Thanks for catching the other one - I ran into that as well while testing, and fixed it in my deploy branch on GH directly. And then deleted that branch 🫣

**Comment by bradygaster** [comment]
> See step 5 of the README - you have to create a secret called `DB_PWD` for PostgreSQL.

AH i missed that.

**Comment by bradygaster** [comment]
Saw this happen once I deployed and the front-end was not responsive. 

![image](https://user-images.githubusercontent.com/41929050/186275405-925f26de-76b6-4ecc-b519-a72c14299fde.png)

**Comment by bradygaster** [comment]
I looked more closely at the code and had an idea, and this aligns really closely with a theme I'm looking at closely and would love to try a solution out - mind if I change your data protection approach to use Key Vault and Azure Storage Blobs instead of the database?

**Comment by joergjo** [comment]
The `DataProtectionKeys` table is created by the container _instance_ that executes the [SQL script generated by EF Core](deploy/migrate.sql). So if this didn't work, changing the data protection implementation won't help, since you will also miss the `Ads` table for the app itself.

Could it be that the Container App deployment was actually faster than the Container Instance deployment that runs the script? I've just looked at the Bicep code, and it doesn't use an explicit dependency between the app modules and the database module - I've never seen the Container Apps deploying _that_ fast...

Another minor flaw I've found: The Bicep database module always sources the SQL script from this very repo: https://github.com/joergjo/contosoads-containerapp/blob/d6aa1cdbcada516510dc4ad02ac1b1f6749f9f37/deploy/modules/database.bicep#L126. 

So I guess I need to extract another parameter here. I'll add these changes later today.

**Comment by joergjo** [comment]
I've added explicit dependencies from the Container Apps modules to the Database module and exposed the repo URL. The screenshots are also up to date.
